heat/core/io.py

```diff
@@ -1193,3 +1195,75 @@ def load_npy_from_path(
 
     x = factories.array(larray, dtype=dtype, device=device, is_split=split, comm=comm)
     return x
+
+
+def load_csv_from_folder(
+    path: str,
+    dtype: datatype = types.int32,
+    split: int = 0,
+    device: Optional[str] = None,
+    comm: Optional[Communication] = None,
+    func: Optional[callable] = None,
+) -> DNDarray:
+    """
+    Loads multiple .csv files into one DNDarray which will be returned. The data will be concatenated along the split axis provided as input.
+
+    Parameters
+    ----------
+    path : str
+        Path to the directory in which .csv-files are located.
+    dtype : datatype, optional
+        Data type of the resulting array.
+    split : int
+        Along which axis the loaded arrays should be concatenated.
+    device : str, optional
+        The device id on which to place the data, defaults to globally set default device.
+    comm : Communication, optional
+        The communication to use for the data distribution, default is 'heat.MPI_WORLD'
+    func : pandas.DataFrame, optional
+        The function the files have to go through before being added to the array.
+    """
+    if not isinstance(path, str):
+        raise TypeError(f"path must be str, not {type(path)}")
+    elif split is not None and not isinstance(split, int):
+        raise TypeError(f"split must be None or int, not {type(split)}")
+
+    process_number = MPI_WORLD.size
+    file_list = []
+    for file in os.listdir(path):
+        if fnmatch.fnmatch(file, "*.csv"):
+            file_list.append(file)
+    n_files = len(file_list)
+
+    if n_files == 0:
+        raise ValueError("No .csv Files were found")
+    if (n_files < process_number) and (process_number > 1):
+        raise RuntimeError("Number of processes can't exceed number of files")
+
+    rank = MPI_WORLD.rank
+    if rank + 1 != process_number:
+        n_for_procs = n_files // process_number
+    else:
+        n_for_procs = (n_files // process_number) + (n_files % process_number)
+
+    local_list = [
```

Comment:
das sieht mir analog zu dem ursprünglichen Code in load-npy aus. Können wir hier die Modifikation von @mtar übernehmen?